allennlp/semparse/worlds/wikitables_variable_free_world.py

```diff
@@ -52,50 +52,108 @@ def __init__(self, table_context: TableQuestionContext) -> None:
                                                  "num": types.NUMBER_TYPE},
                          global_type_signatures=types.COMMON_TYPE_SIGNATURE,
                          global_name_mapping=types.COMMON_NAME_MAPPING)
-        # TODO (pradeep): Do we need constant type prefixes?
         self.table_context = table_context
+        # We add name mapping and signatures corresponding to specific column types to the local
+        # name mapping based on the table content here.
+        column_types = table_context.column_types.values()
+        self._table_has_string_columns = False
+        self._table_has_date_columns = False
+        self._table_has_number_columns = False
+        if "string" in column_types:
+            for name, translated_name in types.STRING_COLUMN_NAME_MAPPING.items():
+                signature = types.STRING_COLUMN_TYPE_SIGNATURE[translated_name]
+                self._add_name_mapping(name, translated_name, signature)
+            self._table_has_string_columns = True
+        if "date" in column_types:
+            for name, translated_name in types.DATE_COLUMN_NAME_MAPPING.items():
+                signature = types.DATE_COLUMN_TYPE_SIGNATURE[translated_name]
+                self._add_name_mapping(name, translated_name, signature)
+            # Adding -1 to mapping because we need it for dates where not all three fields are
+            # specified. We want to do this only when the table has a date column. This is because
+            # the knowledge graph is also constructed in such a way that -1 is an entity with date
+            # columns as the neighbors only if any date columns exist in the table.
+            self._map_name(f"num:-1", keep_mapping=True)
```

Comment:
I think we may want to handle the `-1` in dates separately from the typical "number" entity mechanism here - maybe have it be a global action, not a linked action at all.  But we can worry about that later.